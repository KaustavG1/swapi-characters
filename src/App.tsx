import { Route, Routes } from "react-router";
import Dashboard from "./pages/Dashboard";
import CharacterDetails from "./pages/CharacterDetails";
import Favourites from "./pages/Favourites";
import "./App.css";

function App() {
  return (
    <Routes>
      <Route path="/" element={<Dashboard />} />
      <Route path="/favourites" element={<Favourites />} />
      <Route path=":id" element={<CharacterDetails />} />
    </Routes>
  );
}

export default App;
