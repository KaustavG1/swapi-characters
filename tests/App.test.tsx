import { render, screen } from "@testing-library/react";
import App from "../src/App";

describe("App", () => {
  it("renders loading", () => {
    const { container } = render(<App />);
    const headline = screen.getByText(/Loading/i);
    expect(headline).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("renders headline", () => {
    const { container } = render(<App />);
    const headline = screen.getByText(/Loading/i);
    expect(headline).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
