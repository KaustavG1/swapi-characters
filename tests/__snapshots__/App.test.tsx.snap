// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > renders headline 1`] = `
<div>
  <div
    class="blank-page"
  >
    <div
      class="loader"
    >
      Loading ...
    </div>
  </div>
</div>
`;

exports[`App > renders loading 1`] = `
<div>
  <div
    class="blank-page"
  >
    <div
      class="loader"
    >
      Loading ...
    </div>
  </div>
</div>
`;
